engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/CurrentEngineVersionNode.java

```diff
@@ -0,0 +1,45 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.version.BuildVersion;
+
+@BuiltinMethod(
+    type = "Meta",
+    name = "engine_version",
+    description = "Returns the version of the currently running Enso engine.",
+    autoRegister = false)
+public class CurrentEngineVersionNode extends Node {
+
+  public Text execute() {
+    return getCurrentVersion();
+  }
+
+  @CompilerDirectives.TruffleBoundary
+  private Text getCurrentVersion() {
+    StringBuilder sb = new StringBuilder();
```

Comment:
Exposing text messages as an API is a [well known antipattern](http://wiki.apidesign.org/wiki/Category:APIDesignPatterns:Anti) which has bitten many. See for example _Practical API Design_, Chapter 3: _"Determining what makes a good API"_, page 31:

> Beware of situations where there is no alternative to parsing text messages! If the informa-
tion isn’t available in other ways, people will parse any textual output generated by your code.
For example, this happened to the JDK’s Exception.printStackTrace(OutputStream).

Reply:
Will you be ok if I rename this method to `engine_version_info` or something else?

Reply:
The whole point of this method is to have a single method to get a quick glance of the currently running version.

Just `0.0.0-dev` version string is not enough to get that - it will be the same string for every dev build, regardless of which commit it was build from, so that tells very little.

If I split this up into separate methods, it will be tedious to reconstruct - the whole rationale is to easily get a debug overview.

This method is not meant to be used or parsed by users, if anyone uses it like that - I guess the usage is the anti-pattern. Is the ability for users to abuse a helper debug method enough reason not to have it?

Reply:
> Is the ability for users to abuse a helper debug method enough reason not to have it?

Yes, it is! By exposing a text API like this either of:

- changing the text message
- removing the method

constitutes an **incompatible change** and it is a potential _threat to existing Enso users_. 

In addition to that. In its current form this change doesn't meet requirements for an API change. There is a checkbox:

> [ ] Unit tests have been written where possible.

Where else shall there be tests then when doing an API change?